File: Day076_2025/Q076_Leetcode.java
import java.util.*;

class Q076_Leetcode {
    public boolean divideArray(int[] nums) {
        Set<Integer> unpaired = new HashSet<>();
        
        for (int num : nums) {
            if (unpaired.contains(num)) {
                unpaired.remove(num);
            } else {
                unpaired.add(num);
            }
        }

        return unpaired.isEmpty();
    }
}

File: Day076_2025/Q076_Leetcode.cpp
class Solution {
    public:
        bool divideArray(vector<int>& nums) {
            // Q. Divide Array Into Equal Pairs
            unordered_set<int> unpaired;
            for (int num : nums) {
                if (unpaired.count(num)) {
                    unpaired.erase(num);
                } else {
                    unpaired.insert(num);
                }
            }
    
            return unpaired.empty();
        }
    };